main.tf

```
@@ -162,9 +270,9 @@ resource "aws_ram_resource_association" "this" {
 }
 
 resource "aws_ram_principal_association" "this" {
-  count = var.create_tgw && var.share_tgw ? length(var.ram_principals) : 0
+  for_each = var.create_tgw && var.share_tgw ? var.ram_principals : []
```

```suggestion
  for_each = var.create_tgw && var.share_tgw ? var.ram_principals : toset([])
```

The right side should be a `set`... 

Even fixing it, we will face the `for_each` problem for computed values as mentioned [here](https://github.com/terraform-aws-modules/terraform-aws-transit-gateway/pull/95#discussion_r1098714853) by @bryantbiggs

However, the current version of this code is using `count`which will produce the same error for computed values 🤔